Documentation for Analytics and Consent Management Implementation (#77)
DFE-Digital/improve-school-workload-and-wellbeing: docs/analytics-overview.md

# Website Analytics Implementation Overview

## Introduction

This document provides a high-level overview of how the Improve workload and wellbeing for school staff service utilises
Google Analytics 4 (GA4), Google Tag Manager (GTM), cookies, and a consent banner to track user interactions and comply
with privacy regulations.

## Google Analytics 4 (GA4)

GA4 is the latest iteration of Google's analytics service, providing insights into website traffic, user behavior, and
conversion metrics. We use GA4 to:

> - Monitor key metrics and performance indicators of our digital presence.

Reviewer: ```suggestion
- Monitor key metrics and performance indicators of our service.
```

> - Understand user paths and interactions with our website.

Reviewer: ```suggestion
- Understand user behaviour and interactions with our website.
```

> - Enhance marketing strategies and user experience based on data-driven insights.

Reviewer: ```suggestion
- Improve the service based on data-driven insights.
```

## Google Tag Manager (GTM)

GTM is a tag management system that allows us to quickly and easily update tags and code snippets on our website, including:

- Deployment and management of GA4 tracking codes.
- Configuration of event tracking for user interactions without altering the source code.
- Managing and testing marketing tags and related code snippets.

## Cookies

Cookies are small data files stored on the user's device, which we use to:

> - Remember user preferences and sessions.
> - Collect information that assists with personalizing the user experience.
> - Track user activity for analytics and advertising purposes in compliance with our privacy policy.

Reviewer 2: Making it more applicable to our service:

```suggestion
- Remember user preference for analytics tracking.
- Track user activity for analytics and advertising purposes in compliance with our privacy policy.
```

## Consent Banner

We respect user privacy and comply with legal requirements for data protection. As such, we implement a consent banner that:

- Informs users about the use of cookies and tracking technologies on our website.
- Allows users to give, or deny their consent for data collection.
- Ensures that no tracking cookies are placed without explicit user consent, complying with GDPR, CCPA, and other data
protection laws.

## Consent Management and Data Collection Process

> When a user visits our website, the following occurs:

Reviewer 2: We could also say somewhere in this section that before indicating their consent preference, `consentStatus` is set to `false`.

1. The consent banner is displayed, asking the user for their consent to track their interactions.
2. If the user consents, `consentStatus` is set to `true` in the data layer and GA4 tracking is fully enabled.
3. If the user does not consent, `consentStatus` is set to `false` and GA4 tracking is limited to anonymized, aggregate data.
4. GTM uses the `consentStatus` variable to control the firing of tags based on the user's consent.
5. The website will respect the user's choice, and their consent preference will be stored in cookies for future visits.

## Data Privacy and User Control

Our users have control over their data. They can:

> - Change their consent preferences at any time via the consent banner or website settings.

Reviewer 2: ```suggestion
- Change their consent preferences at any time via the consent banner or cookies policy page.
```
